### ISSUE TITLE
Improve/test/fix image viewer

### ISSUE THREAD

**Issue by windymilla** [DistributedProofreaders/guiguts-py#56]
Image viewer is very rudimentary, allowing pan/zoom by mouse only (drag, Control+mouse wheel, maybe scroll gestures on mousepads)

Also hasn't been tested on different platforms.

At minimum it needs
- Zoom buttons: in & out, fit to page width/height or similar

Also consider what zoom/position to use when new image is loaded, which could be different size/aspect ratio.
Also what to do if floating window is resized.

**Comment by srjfoo** [comment]
On modern Macs, for those who use a trackpad (tested on a MBP M1, not on a desktop with an external trackpad), control plus one of shift, option or cmd and the two-finger drag, zooming in and out works. control plus the two finger drag is a system default for zooming in on the whole screen, or a portion of the screen as set by the system preferences.

I don't know how easy it would be to implement, but certain applications allow "pinch to zoom". For instance, in this browser window, I am allowed to pinch to zoom within the browser window, and it doesn't affect anything else.

It would also be nice to see the ctrl/cmd plus + and - keyboard shortcuts implemented for zooming in and out.

**Comment by srjfoo** [comment]
When testing, I occasionally play with the image resizing currently in place (which is pretty painful on a Mac), and I ponder images in gg2. This time, I actually did some searching, and found [https://stackoverflow.com/questions/63787755/dynamic-resizing-of-image-using-pil-and-tkinter](https://stackoverflow.com/questions/63787755/dynamic-resizing-of-image-using-pil-and-tkinter). It may ultimately not be of use, but I thought I'd mention it, just in case it is when the time comes.

**Comment by windymilla** [comment]
Bug discovered by @srjfoo when auto imaging through file:

> I brought up gg2 in the footnotes branch (making sure ahead of time that the project actually had some footnotes: the first project I tried didn't). Opened the images. I think I resized the image: I usually do. When I do that, the image area generally comes down to where the scrollbars disappear, because I generally size the image to fit inside the image pane/set the image pane size to be no smaller in either dimension than the image.
When I opened the Footnote results list, the image area was immediately resized to max. The screenshot is designed to show that the image pane essentially doubled in height and width (according to the extra space at the botto.m and the lengths of the scrollbars that weren't there before). Actually, more than doubled. I zoomed out far enough that the scrollbars went away again, and the image is about the size of a postage stamp: about a quarter of the width of the image pane.
If I close footnote fixup, and reopen, the background for the image gets even larger, and at some point, I can't scroll out anymore.


```
21:35:10: ERROR - root.py:45 - Tkinter Exception
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.9/Frameworks/Python.framework/Versions/3.11/lib/python3.11/tkinter/__init__.py", line 1967, in __call__
    return self.func(*args)
           ^^^^^^^^^^^^^^^^
  File "/Users/sharonjoiner/dpdev/guiguts-py-3.11/src/guiguts/mainwindow.py", line 373, in wheel_scroll
    self.show_image()
  File "/Users/sharonjoiner/dpdev/guiguts-py-3.11/src/guiguts/mainwindow.py", line 345, in show_image
    image.resize(
  File "/Users/sharonjoiner/Library/Caches/pypoetry/virtualenvs/guiguts-gE8TiuTD-py3.11/lib/python3.11/site-packages/PIL/Image.py", line 2200, in resize
    return self._new(self.im.resize(size, resample, box))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: height and width must be > 0

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.9/Frameworks/Python.framework/Versions/3.11/lib/python3.11/tkinter/__init__.py", line 1967, in __call__
    return self.func(*args)
           ^^^^^^^^^^^^^^^^
  File "/Users/sharonjoiner/dpdev/guiguts-py-3.11/src/guiguts/mainwindow.py", line 345, in show_image
    image.resize(
  File "/Users/sharonjoiner/Library/Caches/pypoetry/virtualenvs/guiguts-gE8TiuTD-py3.11/lib/python3.11/site-packages/PIL/Image.py", line 2200, in resize
    return self._new(self.im.resize(size, resample, box))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: height and width must be > 0

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.9/Frameworks/Python.framework/Versions/3.11/lib/python3.11/tkinter/__init__.py", line 1967, in __call__
    return self.func(*args)
           ^^^^^^^^^^^^^^^^
  File "/Users/sharonjoiner/dpdev/guiguts-py-3.11/src/guiguts/mainwindow.py", line 345, in show_image
    image.resize(
  File "/Users/sharonjoiner/Library/Caches/pypoetry/virtualenvs/guiguts-gE8TiuTD-py3.11/lib/python3.11/site-packages/PIL/Image.py", line 2200, in resize
    return self._new(self.im.resize(size, resample, box))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: height and width must be > 0
```

**Comment by srjfoo** [comment]
Stashing [python-image-processing-libraries/](https://www.geeksforgeeks.org/python-image-processing-libraries/) for future reference.

**Comment by windymilla** [comment]
> I don't know how easy it would be to implement, but certain applications allow "pinch to zoom". For instance, in this browser window, I am allowed to pinch to zoom within the browser window, and it doesn't affect anything else.

According to this, it has been requested, but isn't supported yet: https://core.tcl-lang.org/tips/doc/trunk/tip/570.md

**Comment by windymilla** [comment]
Fixed by #456
